### PR TITLE
add check for new translations when importing, set status to saved..

### PR DESCRIPTION
**Pull request by vsch**
since the translation it is already in the translation file. Otherwise all newly imported translations are marked as changed which is incorrect.

Ignore the export to lang.corr instead of lang. I don't trust anything overwriting my source files and prefer that automatic file creation is done into a parallel directory so that I can merge changes easily with a diff tool. gives me another fail safe in the process.

BTW, great job. I was planning on writing one from scratch to reduce my workload on preparing translations to give to translators and then working their changes into the source but then I came across your package and loved it. Thank you. I will add a translation resource into the package so that the translation manager can be easily localized. Translators would prefer to see the GUI in their own language too.

### CODE REVIEW

File: src/Barryvdh/TranslationManager/Manager.php [hunk 1]
```diff
@@ -137,7 +137,7 @@ public function exportTranslations($group)
             foreach($tree as $locale => $groups){
                 if(isset($groups[$group])){
                     $translations = $groups[$group];
-                    $path = $this->app->make('path').'/lang/'.$locale.'/'.$group.'.php';
+                    $path = $this->app->make('path').'/lang.corr/'.$locale.'/'.$group.'.php';
```
Comment (barryvdh) [inline src/Barryvdh/TranslationManager/Manager.php]: lang.corr?

Comment (vsch) [inline src/Barryvdh/TranslationManager/Manager.php]: I did ask to ignore that little chunk. I was not planning to have it in the pull request but missed it when I made the commit.

To explain why lang.corr. In my case as I am working now, I generate the translation files for all changes coming from translators, using some cobbled together scripts and csv import/export, into lang.corr (language corrections) and then merge them with a diff tool. That way I get to review the changes and don’t lose the comments in the files.

I don’t trust the translations to be overwritten. Sometimes the context is not clear and the corrected translation is wrong.

I’ll try to be more careful with future pull requests. I’m still wobbly with git. I made a separate branch that I will use for changes destined for pull requests. I plan on making lots of customizations specific to the project I am using it on so not all changes will be applicable to the package. I am hoping to catch the extra changes locally by having to merge into a local branch first.

BTW, thank you again for a great job.

Best regards,


Vladimir.

**Comment by barryvdh** [comment]
Why did you close this? What are those other commits?

**Comment by vsch** [comment]
Hi Barry,

I didn’t explicitly close it. I created a branch ‘develop’ to match the branch name in the main project so that PHPStorm can do sync commits with all sub-modules of the main project.

Yesterday, I deleted the ‘devel’ branch, not thinking about the pending pull request from it. The rest is github behaviour from deleting a branch with a pending pull-request.

If you want I can create another pull request for that change.

At the moment my version is too ingrained into the project I am working on, and vice-versa. I hack features as I need them and worry about clean-up and making it separate package later. I will probably have to release my copy as a separate package. There are too many features that won’t appeal to a wide audience, at least until major cleanup and documentation, the changes are not a drop-in replacement for your package, for now it is only for Laravel 4.2. However, some will want them and won’t mind the extra effort.

Here is the current change list, of the top of my head:

Site wide in place editing of translations. When in this mode all translations, except ones used for HTML attribute values, are generated as links for editing, This is not a drop-in feature as it requires cooperation with all the generated HTML views.
Database Only publishing: this mode is used on prod servers in a load balancing group where writing to files is pointless because it will only affect the instance serving the request. Instead, in this mode, ’saved’ cached changes returned for translations requests without having the translation files modified.
Added a few buttons to x-editable to automate common tasks including auto translate (en to ru for now) using Yandex translate; generate plural forms for choice() (automatic for english, semi-automatic using Yandex translate for Russian; capitalize/lowercase word/all words; restore translation to last saved/last published; simulated copy/paste.
Added import group, delete group, delete keys buttons. When logging missing keys groups are created because they are misspelled in the code, delete group is useful for that, same for the delete keys.
Search modal results are editable. i.e. The translations showing up in the search results can be edited in-place instead of having to go to the group. Speeds up corrections / copying of translations from multiple groups.
Dashboard view to show missing/changed translations across all groups.
Yandex translate text boxes on main view to do quick translations right from translation manager, currently using only English <—> Russian.
Missing keys tracking for production using a 1:n lottery. Setting the config value to n will randomly pick 1 out of n sessions to enable missing key tracking for that session. That way missing keys can be tracked in production but limit the performance impact to 1 out of n users.
Included management of overrides of package translations in translation manager. To disable add the package/group to the excluded group list in config.
Translation parameters to allow unpublished translations to be returned as a result of trans() or choice(). Used for previewing translation changes before publishing them site-wide.
Clean-up translation export so that the generated code is easier on the eyes. I hate looking at scrambled formatting.

I won’t have time to clean-up the code for a while. I am doing 16 to 18 hour days, 7 days a week on the project. So as much as I would like to clean up and document, I just won’t have the time until after launch.

Best regards,


Vladimir.

**Pull request closed without merging**